lib/tasks/gettext.rake

```diff
@@ -20,15 +20,23 @@ begin
     end
   end
 
-  desc 'Extract plugin strings - called via rake plugin:gettext[plugin_name]'
-  task 'plugin:gettext', :engine do |t, args|
-    @domain = args[:engine]
-    @engine = "#{@domain.camelize}::Engine".constantize
-    @engine_root = @engine.root
+  desc 'Extract plugin strings - called via rake plugin:gettext[engine]'
+  task 'plugin:gettext', [:engine] => [:environment] do |t, args|
+    unless args[:engine]
+      puts "You must specify the name of the plugin (e.g. rake plugin:gettext['my_plugin'])"
+      exit 1
+    end
+    path_name = args[:engine].tr('-', '_')
+    @engine = "#{path_name.camelize}::Engine".constantize
+
+    @plugin = Foreman::Plugin.find(args[:engine])
+    @domain = @plugin.gettext_domain
+
+    raise "Plugin '#{@plugin.name}' does not have translations registered'" unless @domain
```

Comment:
This doesn't have any fallback, so plugins need to start using `register_gettext_domain`. That's good, but if they want to remain compatible with Foreman < 3.7 then what would you recommend? `respond_to?(:register_gettext_domain)`?

Reply:
Probably just making the new way/old way decision based on version of Foreman